controllers/cards.js

```diff
@@ -1,41 +1,41 @@
+const NotFoundError = require('../errors/not-found-err');
+const ForbiddenError = require('../errors/forbidden-err');
 const Card = require('../models/card');
 
 // Возвращает все карточки
-const returnAllcards = (req, res) => {
+const returnAllcards = (req, res, next) => {
   Card.find({})
     .then(card => res.send({ data: card }))
-    .catch(err => res.status(500).send({ message: 'Карточки не загружены', error: err.message }));
+    .catch(next);
 };
 
 // Удаляет карточку по идентификатору
-const removeCardId = (req, res) => {
+const removeCardId = (req, res, next) => {
   Card.findById(req.params.cardId)
     .then(card => {
       if (!card) {
-        res.status(404).send({ message: `Нет карточки с id ${req.params.cardId}` });
-        return;
+        throw new NotFoundError(`Нет карточки с id ${req.params.cardId}`);
       }
       if (!card.owner.equals(req.user._id)) {
-        res.status(403).send({ message: 'Вы не можете удалить карточку, созданную другим пользователем' });
-        return;
+        throw new ForbiddenError('Вы не можете удалить карточку, созданную другим пользователем');
       }
       Card.deleteOne(card)
         .then(() => res.send({ data: card }));
```

Comment:
Можно лучше: использовать единый `.catch()` блок для вложенных промисов.
Особенность работы промисов в том, что `.then()` блоки выполняются в том порядке, в котором они были описаны. Причем вызов следующего `.then()` запускается только после того как промис перейдет в состояние выполнен (_resolved_) или как текущий `.then()` блок будет выполнен. Это означает, что вложенный промис и его `.then()` блоки тоже могут быть встроены в цепочку вызовов, если внутри `.then()` блока будет выполнен `return`.
При наличии одной цепочки вложенный `.catch()` блок можно убирать, потому что любое отклонение промиса (_reject_) будет поймано верхнеуровневым `.catch()` блоком.

```suggestion
      return Card.deleteOne(card)
        .then(() => res.send({ data: card }));
```